src/pages/CategoryProduct/index.jsx

```diff
@@ -1,11 +1,86 @@
-import React from 'react';
+import Container from '@material-ui/core/Container';
+import Grid from '@material-ui/core/Grid';
+import React, { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory, useParams } from 'react-router-dom';
+import Aside from '../../components/Aside';
+import ContentHeader from '../../components/ContentHeader';
+import ListItem from '../../components/ListItem';
+import LoadingProduct from '../../components/LoadingProduct';
+import Paginate from '../../components/Paginate';
+import Sort from '../../components/Sort';
+import sortBy from '../../utils/sortBy';
+import * as actions from './../../actions';
 
 function CategoryProduct() {
+    const { t } = useTranslation();
+    const dispatch = useDispatch();
+    const history = useHistory();
+    const { categoryId, pageId } = useParams();
 
-    return (
-        <div>
+    const loading = useSelector(state => state.loading);
+    const products = useSelector(state => state.products);
+    const filterProducts = useSelector(state => state.filterProduct.filterProductData);
+    const limit = useSelector(state => state.paginate.limit);
+    const page = useSelector(state => state.paginate.page);
+
+    const totalPage = Math.ceil(filterProducts.length / limit);
+    const indexOfLastList = page * limit;
+    const indexOfFirstList = indexOfLastList - limit;
+    const currentList = filterProducts.slice(indexOfFirstList, indexOfLastList);
+
+    useEffect(() => {
+        document.title = t(`${categoryId.replace("-", " ")}`);
+        dispatch(actions.actChangePage(parseInt(pageId)));
+        dispatch(actions.actShowLoading());
+
+        const categoryProducts = products.filter(product => {
+            return product.categoryId === categoryId;
+        })
+
+        const loadingProduct = setTimeout(() => {
+            dispatch(actions.actChangeFilterProductData(categoryProducts));
+            dispatch(actions.actHideLoading());
+        }, 1500);
 
-        </div>
+        return () => {
+            dispatch(actions.actChangePage(1));
+            clearTimeout(loadingProduct);
+        }
+    }, [categoryId, products]);
+
+    useEffect(() => {
+        history.push(`/product/${categoryId}/page=${page}`);
```

Comment:
sao cái này k dùng link to e

Reply:
cái này e muốn mỗi lần page thay đổi, hay categoryId thay đổi thì e chỉ thay đổi đường dẫn url thôi a.